packages/components/nodes/Mailchimp/Mailchimp.ts

```diff
@@ -0,0 +1,149 @@
+                type: 'options',
+                options: [
+                    {
+                        label: 'Get list of campaigns',
+                        name: 'listCampaigns',
+                        description: 'Returns the list of campaigns'
+                    },
+                    {
+                        label: 'Get campaign',
+                        name: 'getCampaign',
+                        description: 'Return single campaign'
+                    },
+                    {
+                        label: 'Delete campaign',
+                        name: 'deleteCampaign',
+                        description: 'It will delete campaigns'
+                    }
+                ],
+                default: 'listCampaigns'
+            }
+        ] as INodeParams[]
+        this.credentials = [
+            // credentialMethod is mandatory field
+            {
+                label: 'Credential Method',
+                name: 'credentialMethod',
+                type: 'options',
+                options: [
+                    {
+                        label: 'Mailchimp Credentials',
+                        name: 'mailChimpCredential'
+                    }
+                ],
+                default: 'mailChimpCredential'
+            }
+        ] as INodeParams[]
+        this.inputParameters = [
+            {
+                label: 'Campaign Id',
+                name: 'campaignId',
+                type: 'string',
+                show: {
+                    'actions.api': ['deleteCampaign', 'getCampaign']
+                },
+                default: ''
+            }
+        ] as INodeParams[]
+    }
+
+    async run(nodeData: INodeData): Promise<INodeExecutionData[] | null> {
+        // function to make calls
+        let authObj: Auth
+        async function makeApiCall(method: string, url: string): Promise<any[]> {
+            const axiosConfig: AxiosRequestConfig = {
+                method: method as Method,
+                url,
+                headers: { 'Content-Type': 'application/json' },
+                auth: {
+                    ...authObj
+                }
+            }
+            let responseData: any[] = []
+            try {
+                const response = await axios(axiosConfig)
+                if (response?.data) {
+                    responseData.push(response?.data)
+                }
+            } catch (err) {
+                throw handleErrorMessage(err)
+            }
+            return responseData
+        }
+        // function to start running the node
+        const actionData = nodeData.actions
+        const credentials = nodeData.credentials
+        if (actionData === undefined || credentials === undefined) {
+            throw handleErrorMessage({ message: 'Required data missing' })
+        }
+        const api = actionData.api as string
+        const apiKey = credentials.apiKey as string
+        const dc = credentials.dc as string
```

Comment:
we can extract dc from API key as documented - It’s also appended to your API key in the form key-dc; if your API key is 0123456789abcdef0123456789abcde-us6, then the data center subdomain is us6.